packages/next/src/server/app-render/work-unit-async-storage.external.ts

```diff
@@ -189,30 +196,40 @@ export function getExpectedRequestStore(
   callingExpression: string
 ): RequestStore {
   const workUnitStore = workUnitAsyncStorageInstance.getStore()
-  if (workUnitStore) {
-    if (workUnitStore.type === 'request') {
+
+  if (!workUnitStore) {
+    throwForMissingRequestStore(callingExpression)
+  }
+
+  switch (workUnitStore.type) {
+    case 'request':
       return workUnitStore
-    }
-    if (
-      workUnitStore.type === 'prerender' ||
-      workUnitStore.type === 'prerender-ppr' ||
-      workUnitStore.type === 'prerender-legacy'
-    ) {
+
+    case 'prerender':
+    case 'prerender-ppr':
+    case 'prerender-legacy':
       // This should not happen because we should have checked it already.
       throw new Error(
         `\`${callingExpression}\` cannot be called inside a prerender. This is a bug in Next.js.`
       )
-    }
-    if (workUnitStore.type === 'cache') {
+
+    case 'cache':
       throw new Error(
         `\`${callingExpression}\` cannot be called inside "use cache". Call it outside and pass an argument instead. Read more: https://nextjs.org/docs/messages/next-request-in-use-cache`
       )
-    } else if (workUnitStore.type === 'unstable-cache') {
+
+    case 'unstable-cache':
       throw new Error(
         `\`${callingExpression}\` cannot be called inside unstable_cache. Call it outside and pass an argument instead. Read more: https://nextjs.org/docs/app/api-reference/functions/unstable_cache`
       )
-    }
+
+    default:
+      const _exhaustiveCheck: never = workUnitStore
+      return _exhaustiveCheck
```

Comment:
This hurts, but I understand.

Reply:
I was also surprised to learn that this is actually the [officially recommended way](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking) to do this. 😒